packages/file-explorer/src/FileExplorer.test.tsx

```diff
@@ -0,0 +1,193 @@
+  });
+
+  it('should be able to rename a directory', async () => {
+    await clickContextMenuButton(/testdir0/i, /rename/i);
+
+    await user.keyboard('test{enter}');
+    expect(onRename).toBeCalledWith('/testdir0/', '/testdir0test/');
+  });
+
+  it('should throw an error when renaming a file as a file that already exists', async () => {
+    await clickContextMenuButton(/testfile0/i, /rename/i);
+
+    await user.keyboard('{Backspace}{1}{Enter}');
+    expect(
+      await screen.findByText('Error: Name already exists')
+    ).toBeInTheDocument();
+    expect(onRename).not.toBeCalledWith('/testfile0', '/testfile1');
+  });
+
+  it('should open a modal when deleting an item', async () => {
+    await clickContextMenuButton(/testfile0/i, /delete/i);
+    expect(
+      await screen.findByText(
+        /Are you sure you want to delete "\/testfile0"\?/i
+      )
+    );
+  });
+
+  it('should close the modal on cancelling the delete', async () => {
+    await clickContextMenuButton(/testfile0/i, /delete/i);
+
+    const cancelButton = await screen.findByRole('button', { name: /cancel/i });
+    expect(cancelButton).toBeInTheDocument();
+    await user.click(cancelButton);
+
+    await waitForElementToBeRemoved(cancelButton);
+    expect(cancelButton).not.toBeInTheDocument();
+  });
+
+  it('should delete the item upon confirmation', async () => {
+    await clickContextMenuButton(/testfile0/i, /delete/i);
+
+    const modalDeleteButton = await screen.findByRole('button', {
+      name: /delete/i,
+    });
+    await user.click(modalDeleteButton);
+    await waitForElementToBeRemoved(modalDeleteButton);
+    expect(onDelete).toBeCalledTimes(1);
+  });
+});
+
+// Need to implement tests for drag and drop
+
+/* describe('drag and drop works properly', () => {
+  let user: ReturnType<typeof userEvent.setup>;
+  let dirs: DirectoryStorageItem[] = [];
+  let files: FileStorageItem[] = [];
+  let items: FileStorageItem[] = [];
+  const onRename = jest.fn();
+  const onDelete = jest.fn();
+
+  beforeEach(async () => {
+    user = userEvent.setup();
+    dirs = makeDirectories();
+    files = makeFiles();
+    files.push(makeNested([2], 2));
+    files.push(makeNested([0, 3], 4));
+    items = dirs.concat(files);
+
+    const fileStorage = new MockFileStorage(items);
+    renderFileExplorer({ storage: fileStorage, onRename, onDelete });
+    const foundItems = await screen.findAllByRole('listitem');
+    expect(foundItems).toHaveLength(items.length);
+  });
+
+  it('should drag a file into a directory', async () => {
+    const file = await screen.findByText('testfile0');
+    const directory = await screen.findByText('testdir0');
+    // await user.pointer({ keys: '[MouseLeft>]', target: file });
+    fireEvent.mouseDown(file);
+    fireEvent.mouseMove(directory);
+    fireEvent.mouseUp(directory);
+    // await user.pointer({ keys: '[/MouseLeft]' });
+
+    // expect(file).not.toBeInTheDocument();
+  });
+});
+ */
```

Comment:
Remove commented out code. We can make a ticket for following up on creating drag 'n' drop tests in FileExplorer, and include this snippet in the ticket. And just have a comment here `TODO: #XXX` with the ticket number.

Ultimately the drag and drop tests may end up being done in Playwright instead of jest though.